lib/ansiblelint/testing/__init__.py

```diff
@@ -31,12 +31,11 @@ def _call_runner(self, path) -> List["MatchError"]:
 
     def run_playbook(self, playbook_text):
         """Lints received text as a playbook."""
-        play_root = tempfile.mkdtemp()
-        with open(os.path.join(play_root, 'playbook.yml'), 'w') as fp:
+        with tempfile.NamedTemporaryFile(mode="w", suffix=".yml", prefix="playbook") as fp:
             fp.write(playbook_text)
-        results = self._call_runner(fp.name)
-        shutil.rmtree(play_root)
-        return results
+            fp.flush()
+            results = self._call_runner(fp.name)
+            return results
```

Comment:
Only `self._call_runner()` needs to be inside of CM, let's return after the tmp dir has been cleaned up:
```suggestion
        return results
```